query/aql_compiler.go

```diff
@@ -280,6 +280,7 @@ func (qc *AQLQueryContext) matchGeoJoin(joinTableID int, mainTableSchema *memsto
 		pointTableID:  point.TableID,
 		pointColumnID: point.ColumnID,
 		dimIndex:      -1,
+		inOrOut:       true,
```

Comment:
if we separate intersect and dimension write, we might need to add compiler check and fail geo join queries with inOrOut as false

Reply:
We already fail it. 
see aql_compiler:677

if isCallRef && callRef.Name == geographyIntersectsCallName {
				qc.Error = utils.StackError(nil, "Not %s condition is not allowed", geographyIntersectsCallName)
				break
			}

Reply:
I see.